core/vm/contracts.go

```diff
@@ -106,13 +57,76 @@ var PrecompiledContractsBLS = map[common.Address]PrecompiledContract{
 }
 
 var (
-	PrecompiledAddressesBerlin    []common.Address
-	PrecompiledAddressesIstanbul  []common.Address
-	PrecompiledAddressesByzantium []common.Address
-	PrecompiledAddressesHomestead []common.Address
+	PrecompiledAddressesBerlin     []common.Address
+	PrecompiledAddressesMiko       []common.Address
+	PrecompiledAddressesConsortium []common.Address
+	PrecompiledAddressesIstanbul   []common.Address
+	PrecompiledAddressesByzantium  []common.Address
+	PrecompiledAddressesHomestead  []common.Address
+
+	// PrecompiledContractsHomestead contains the default set of pre-compiled Ethereum
+	// contracts used in the Frontier and Homestead releases.
+	PrecompiledContractsHomestead = map[common.Address]PrecompiledContract{
+		common.BytesToAddress([]byte{1}): &ecrecover{},
+		common.BytesToAddress([]byte{2}): &sha256hash{},
+		common.BytesToAddress([]byte{3}): &ripemd160hash{},
+		common.BytesToAddress([]byte{4}): &dataCopy{},
+	}
+
+	// PrecompiledContractsByzantium contains the default set of pre-compiled Ethereum
+	// contracts used in the Byzantium release.
+	PrecompiledContractsByzantium map[common.Address]PrecompiledContract
+
+	// PrecompiledContractsIstanbul contains the default set of pre-compiled Ethereum
+	// contracts used in the Istanbul release.
+	PrecompiledContractsIstanbul map[common.Address]PrecompiledContract
+
+	// PrecompiledContractsConsortium contains additional Consortium precompiled contract
+	// beside PrecompiledContractsIstanbul
+	PrecompiledContractsConsortium map[common.Address]PrecompiledContract
+
+	// PrecompiledContractsConsortium contains proof of possession precompiled contract
+	// beside PrecompiledContractsConsortium
+	PrecompiledContractsConsortiumMiko map[common.Address]PrecompiledContract
+
+	// PrecompiledContractsBerlin contains the default set of pre-compiled Ethereum
+	// contracts used in the Berlin release.
+	PrecompiledContractsBerlin map[common.Address]PrecompiledContract
 )
 
+func copyPrecompiledContract(input map[common.Address]PrecompiledContract) map[common.Address]PrecompiledContract {
+	output := make(map[common.Address]PrecompiledContract)
+
+	for address, contract := range input {
+		output[address] = contract
```

Comment:
naming, maybe 
```
input: src 
output: dst
```

Reply:
I've changed to different name. I don't like src and dst as it feels like a move operation.